backend/src/routes/v2/secrets.ts

```diff
@@ -0,0 +1,177 @@
+import express from 'express';
+const router = express.Router();
+import {
+    requireAuth,
+    requireWorkspaceAuth,
+    requireSecretsAuth,
+    validateRequest
+} from '../../middleware';
+import { query, check, body } from 'express-validator';
+import { secretsController } from '../../controllers/v2';
+import { 
+    ADMIN, 
+    MEMBER,
+    SECRET_PERSONAL,
+    SECRET_SHARED
+} from '../../variables';
+
+router.post(
+    '/',
+    body('workspaceId').exists().isString().trim(),
+    body('environment').exists().isString().trim().isIn(['dev', 'staging', 'prod', 'test']),
```

Comment:
So what about custom envrionment names?

Reply:
I think we don't have customer environment names yet. But agree, when we implement that we should change that

Reply:
Yeah Akhil will add these right after.